h2o-core/src/main/java/water/udf/CDistributionFunc.java

```diff
@@ -0,0 +1,86 @@
+package water.udf;
+
+import java.io.Serializable;
+
+/**
+ * Custom Distribution Function
+ */
+public interface CDistributionFunc extends CFunc, Serializable {
```

Comment:
Might make sense to split into 3-4 logical classes.

Reply:
Yeah, I'm still working on the structure...